### PR TITLE
chore: Speed up preprocessing for rosbag loading in Rosbag2ToNonAnnotatedT4Converter

### PR DESCRIPTION
**Pull request by miursh**
## Description
- Simplify preprocessing by skipping full traversal of rosbags, improving processing speed in Rosbag2ToNonAnnotatedT4Converter.
- Fix typo (TracingConversion -> TrackingConversion)

## How to test
Please verify that the pytest passes.

### CODE REVIEW

File: perception_dataset/rosbag2/rosbag2_reader.py [hunk 1]
```diff
@@ -88,6 +96,10 @@ def _set_tf_buffer(self):
     def _set_camera_info(self):
         """set /camera_info to self.camera_info"""
         for topic_name, message in self.read_camera_info():
+            if all(cam_info is not None for cam_info in self.camera_info.values()) and all(
+                frame_id is not None for frame_id in self.sensor_topic_to_frame_id.values()
+            ):
```
Comment (kminoda) [inline perception_dataset/rosbag2/rosbag2_reader.py:101]:
```suggestion
            cam_info_available: bool = all(cam_info is not None for cam_info in self.camera_info.values())
            frame_id_available: bool = all(frame_id is not None for frame_id in self.sensor_topic_to_frame_id.values())
            if cam_info_available and frame_id_available:
```
nits: How about this?

**Review by kminoda — COMMENTED** [review]
Thank you!

File: perception_dataset/rosbag2/rosbag2_reader.py (hunk 1 above)
Comment (miursh) [reply perception_dataset/rosbag2/rosbag2_reader.py:101]: Thank you!

**Review by kminoda — APPROVED** [review]
LGTM!

**Pull request merged**